### ISSUE TITLE
Unable to build project after upgrading realm-gradle-plugin  to version '4.3.0'

### ISSUE THREAD

**Issue by joshpavlovich** [realm/realm-java#5593]
**Bug**: When trying to assemble my project from the command line (gradle clean assembleDebug) the build fails with the following error in the **transformClassesWithRealmTransformerForDebug** task:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':data:transformClassesWithRealmTransformerForDebug'.
> Bad <init> method call from inside of a branch
  Exception Details:
    Location:
      io/realm/transformer/ManagedClassPool.<init>(Ljava/util/Collection;Ljava/util/Collection;)V @59: invokespecial
    Reason:
      Error exists in the bytecode
    Bytecode:
      0x0000000: b800 184e 04bd 001a 5903 0153 5910 ff12
      0x0000010: 04b8 0020 2a5f ab00 0000 0050 0000 0003
      0x0000020: 0000 9b75 0000 0022 0013 85ff 0000 002b
      0x0000030: 159a c03f 0000 003b 5f5a 57b7 0023 a700
      0x0000040: 325f 5a59 0332 b800 295f 57b7 002c a700
      0x0000050: 225f 5a59 0332 1204 b800 30c0 0004 5f57
      0x0000060: b700 33a7 000d bb00 3559 1237 b700 3abf
      0x0000070: 572d 123b 3212 3db9 0043 0200 3a04 1904
      0x0000080: 1245 b800 30c0 0045 2a5f b500 4719 0457
      0x0000090: 2ab6 004b 3a05 1905 2a5f b500 4d19 0557
      0x00000a0: 2d12 4e32 2ab9 0052 0200 572d 1253 322b
      0x00000b0: bb00 5559 2a2a b700 58b9 005c 0300 572d
      0x00000c0: 125d 322c bb00 5f59 2a2a b700 60b9 005c
      0x00000d0: 0300 57b1
    Stackmap Table:
      full_frame(@56,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@65,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@81,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@102,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@112,{Object[#2],Object[#103],Object[#103],Object[#105]},{Object[#107]})

I was also able to reproduce this same build error with the realm-java kotlinExample project.

**Steps to reproduce**

1. git clone git@github.com:realm/realm-java.git
2. cd realm-java/examples/kotlinExample
3. git checkout v4.3.0
4. gradle clean assembleDebug

**Result**
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kotlinExample:transformClassesWithRealmTransformerForDebug'.
> Bad <init> method call from inside of a branch
  Exception Details:
    Location:
      io/realm/transformer/ManagedClassPool.<init>(Ljava/util/Collection;Ljava/util/Collection;)V @59: invokespecial
    Reason:
      Error exists in the bytecode
    Bytecode:
      0x0000000: b800 184e 04bd 001a 5903 0153 5910 ff12
      0x0000010: 04b8 0020 2a5f ab00 0000 0050 0000 0003
      0x0000020: 0000 9b75 0000 0022 0013 85ff 0000 002b
      0x0000030: 159a c03f 0000 003b 5f5a 57b7 0023 a700
      0x0000040: 325f 5a59 0332 b800 295f 57b7 002c a700
      0x0000050: 225f 5a59 0332 1204 b800 30c0 0004 5f57
      0x0000060: b700 33a7 000d bb00 3559 1237 b700 3abf
      0x0000070: 572d 123b 3212 3db9 0043 0200 3a04 1904
      0x0000080: 1245 b800 30c0 0045 2a5f b500 4719 0457
      0x0000090: 2ab6 004b 3a05 1905 2a5f b500 4d19 0557
      0x00000a0: 2d12 4e32 2ab9 0052 0200 572d 1253 322b
      0x00000b0: bb00 5559 2a2a b700 58b9 005c 0300 572d
      0x00000c0: 125d 322c bb00 5f59 2a2a b700 60b9 005c
      0x00000d0: 0300 57b1
    Stackmap Table:
      full_frame(@56,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@65,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@81,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@102,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@112,{Object[#2],Object[#103],Object[#103],Object[#105]},{Object[#107]})

#### Version of Realm and tooling
Realm version(s): 4.3.0

Realm sync feature enabled: no

Android Studio version: 3.0.1

Gradle version: 4.3.1

**Comment by Zhuinden** [comment]
Possibly a side-effect of https://github.com/realm/realm-java/pull/5523 but I don't see what is broken in it.

**Comment by Paul75** [comment]
For me I have a problem when build project it failed, tell me : 
`Error:com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\53, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\0.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\1.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\2.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\3.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\4.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\5.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\6.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\7.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\8.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\9.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\10.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\11.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\12.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\13.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\14.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\15.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\16.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\17.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\18.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\19.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\20.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\21.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\22.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\23.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\24.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\25.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\26.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\27.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\28.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\29.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\30.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\31.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\32.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\33.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\34.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\35.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\36.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\37.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\38.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\39.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\40.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\41.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\42.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\43.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\44.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\45.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\46.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\47.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\48.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\49.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\50.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\51.jar, C:\projets\GOMPOST\GOMPOST_Android\app\build\intermediates\transforms\dexBuilder\debug\52.jar`

Thanks

**Comment by cmelchior** [comment]
@joshpavlovich what version of Java do you have installed?

@Zhuinden No, that shouldn't be related. The bug that there was  about files not being deleted

@Paul75 That doesn't look Realm related. Try doing a full clean and build again?

**Comment by Paul75** [comment]
@cmelchior yes i do a clean and build. With the 4.2.0 no problems to build and run ....

**Comment by Paul75** [comment]
@cmelchior also I have : Error:com.android.tools.r8.errors.CompilationError: Program type already present: org.intellij.lang.annotations.Identifier
and Error:Execution failed for task ':app:transformDexArchiveWithDexMergerForDebug'.
> com.android.build.api.transform.TransformException: java.lang.RuntimeException: java.lang.RuntimeException: com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:

**Comment by Paul75** [comment]
I retest to clean package and why i found this ;:

Error:Could not find com.android.tools.build:gradle:3.1.0-alpha03.
Searched in the following locations:
    file:/C:/Program Files/Android/Android Studio/gradle/m2repository/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.pom
    file:/C:/Program Files/Android/Android Studio/gradle/m2repository/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.jar
    https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.pom
    https://jcenter.bintray.com/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.jar
    https://repo1.maven.org/maven2/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.pom
    https://repo1.maven.org/maven2/com/android/tools/build/gradle/3.1.0-alpha03/gradle-3.1.0-alpha03.jar

**Comment by joshpavlovich** [comment]
@cmelchior I am running Java 8 and macOS High Sierra. 

**Additional Details**

macOS High Sierra, version 10.13.1

java version "1.8.0_20"
Java(TM) SE Runtime Environment (build 1.8.0_20-b26)
Java HotSpot(TM) 64-Bit Server VM (build 25.20-b23, mixed mode)

**Comment by joshpavlovich** [comment]
@cmelchior I updated my JDK to the latest Java 8 version ("1.8.0_152") and now I am able to build (gradle clean assembleDebug) my project and kotlinExample (realm-java/examples/kotlinExample) successfully on my macbook pro.

FYI, I also tested this successfully with realm-java version "4.3.1".

**Comment by Paul75** [comment]
for me now works with 4.3.1 without change version of java ....

**Comment by Zhuinden** [comment]
@cmelchior actually, I've just added Realm 4.3.1 to a completely new project and managed to run into the same thing.

I had JDK 1.8.0_25 but now I updated to _151 and it still does it, I'll try a system reboot soon hoping that'll do something.

Anyways, here is the full stack trace:

```
:app:transformClassesWithRealmTransformerForDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithRealmTransformerForDebug'.
> Bad <init> method call from inside of a branch
  Exception Details:
    Location:
      io/realm/transformer/ManagedClassPool.<init>(Ljava/util/Collection;Ljava/util/Collection;)V @59: invokespecial
    Reason:
      Error exists in the bytecode
    Bytecode:
      0x0000000: b800 184e 04bd 001a 5903 0153 5910 ff12
      0x0000010: 04b8 0020 2a5f ab00 0000 0050 0000 0003
      0x0000020: 0000 9b75 0000 0022 0013 85ff 0000 002b
      0x0000030: 159a c03f 0000 003b 5f5a 57b7 0023 a700
      0x0000040: 325f 5a59 0332 b800 295f 57b7 002c a700
      0x0000050: 225f 5a59 0332 1204 b800 30c0 0004 5f57
      0x0000060: b700 33a7 000d bb00 3559 1237 b700 3abf
      0x0000070: 572d 123b 3212 3db9 0043 0200 3a04 1904
      0x0000080: 1245 b800 30c0 0045 2a5f b500 4719 0457
      0x0000090: 2ab6 004b 3a05 1905 2a5f b500 4d19 0557
      0x00000a0: 2d12 4e32 2ab9 0052 0200 572d 1253 322b
      0x00000b0: bb00 5559 2a2a b700 58b9 005c 0300 572d
      0x00000c0: 125d 322c bb00 5f59 2a2a b700 60b9 005c
      0x00000d0: 0300 57b1                              
    Stackmap Table:
      full_frame(@56,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@65,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@81,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@102,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
      full_frame(@112,{Object[#2],Object[#103],Object[#103],Object[#105]},{Object[#107]})


* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:transformClassesWithRealmTransformerForDebug'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.VerifyError: Bad <init> method call from inside of a branch
Exception Details:
  Location:
    io/realm/transformer/ManagedClassPool.<init>(Ljava/util/Collection;Ljava/util/Collection;)V @59: invokespecial
  Reason:
    Error exists in the bytecode
  Bytecode:
    0x0000000: b800 184e 04bd 001a 5903 0153 5910 ff12
    0x0000010: 04b8 0020 2a5f ab00 0000 0050 0000 0003
    0x0000020: 0000 9b75 0000 0022 0013 85ff 0000 002b
    0x0000030: 159a c03f 0000 003b 5f5a 57b7 0023 a700
    0x0000040: 325f 5a59 0332 b800 295f 57b7 002c a700
    0x0000050: 225f 5a59 0332 1204 b800 30c0 0004 5f57
    0x0000060: b700 33a7 000d bb00 3559 1237 b700 3abf
    0x0000070: 572d 123b 3212 3db9 0043 0200 3a04 1904
    0x0000080: 1245 b800 30c0 0045 2a5f b500 4719 0457
    0x0000090: 2ab6 004b 3a05 1905 2a5f b500 4d19 0557
    0x00000a0: 2d12 4e32 2ab9 0052 0200 572d 1253 322b
    0x00000b0: bb00 5559 2a2a b700 58b9 005c 0300 572d
    0x00000c0: 125d 322c bb00 5f59 2a2a b700 60b9 005c
    0x00000d0: 0300 57b1                              
  Stackmap Table:
    full_frame(@56,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
    full_frame(@65,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
    full_frame(@81,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
    full_frame(@102,{UninitializedThis,Object[#103],Object[#103],Object[#105]},{Object[#107],UninitializedThis})
    full_frame(@112,{Object[#2],Object[#103],Object[#103],Object[#105]},{Object[#107]})

	at io.realm.transformer.RealmTransformer.transform(RealmTransformer.groovy:90)
	at com.android.build.api.transform.Transform.transform(Transform.java:288)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:222)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:218)
	at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:213)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$IncrementalTaskAction.doExecute(DefaultTaskClassInfoStore.java:173)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 27 more
```

**Comment by Zhuinden** [comment]
**Okay I've figured out that just installing latest Java 8 was not enough because I had JDK_HOME set to jdk8_u20**, so I changed the JDK_HOME, deleted my old Java version entirely, re-opened Android Studio and it nagged me to use a new JDK.

Now it works.

BTW this happened to https://github.com/KeepSafe/dexcount-gradle-plugin/issues/12 and https://stackoverflow.com/questions/25428317/bad-init-method-call-from-inside-of-a-branch and  https://stackoverflow.com/questions/25621407/gradle-bad-init-method-call-from-inside-of-a-branch-invokespecial and https://zeroturnaround.com/forums/topic/verifyerror-bad-method-call-from-inside-of-a-branch/

Basically it was a JDK error, and an update to JDK fixes it.
